stock_quant_manual_assign/wizard/assign_manual_quants.py

```diff
@@ -0,0 +1,230 @@
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools.float_utils import float_compare
+
+
+class AssignManualQuants(models.TransientModel):
+    _name = "assign.manual.quants"
+    _description = "Assign Manual Quants"
+
+    @api.constrains("quants_lines")
+    def _check_qty(self):
+        precision_digits = self.env["decimal.precision"].precision_get(
+            "Product Unit of Measure"
+        )
+        for record in self.filtered("quants_lines"):
+            if (
+                float_compare(
+                    record.lines_qty,
+                    record.move_id.product_qty,
+                    precision_digits=precision_digits,
+                )
+                > 0
+            ):
+                raise ValidationError(_("Quantity is higher than the needed one"))
+
+    @api.depends("move_id", "quants_lines", "quants_lines.qty")
+    def _compute_qties(self):
+        for record in self:
+            record.lines_qty = sum(
+                record.quants_lines.filtered("selected").mapped("qty")
+            )
+            record.move_qty = record.move_id.product_qty - record.lines_qty
+
+    lines_qty = fields.Float(
+        string="Reserved qty",
+        compute="_compute_qties",
+        digits="Product Unit of Measure",
+    )
+    move_qty = fields.Float(
+        string="Remaining qty",
+        compute="_compute_qties",
+        digits="Product Unit of Measure",
+    )
+    quants_lines = fields.One2many(
+        "assign.manual.quants.lines", "assign_wizard", string="Quants"
+    )
+    move_id = fields.Many2one(comodel_name="stock.move", string="Move",)
+
+    def assign_quants(self):
+        move = self.move_id
+        move._do_unreserve()
+        for line in self.quants_lines:
+            line._assign_quant_line()
+        # Auto-fill all lines as done
+        for ml in move.move_line_ids:
+            ml.qty_done = ml.product_qty
+        move._recompute_state()
+        move.mapped("picking_id")._compute_state()
+        return {}
+
+    @api.model
+    def default_get(self, fields):
+        res = super(AssignManualQuants, self).default_get(fields)
+        move = self.env["stock.move"].browse(self.env.context["active_id"])
+        available_quants = self.env["stock.quant"].search(
+            [
+                ("location_id", "child_of", move.location_id.id),
+                ("product_id", "=", move.product_id.id),
+                ("quantity", ">", 0),
+            ]
+        )
+        quants_lines = []
+        for quant in available_quants:
+            line = {}
+            line["quant_id"] = quant.id
+            line["on_hand"] = quant.quantity
+            line["location_id"] = quant.location_id.id
+            line["lot_id"] = quant.lot_id.id
+            line["package_id"] = quant.package_id.id
+            line["owner_id"] = quant.owner_id.id
+            line["selected"] = False
```

Comment:
```
line = {
    "quant_id": quant.id,
    "on_hand": quant.quantity,
    "location_id": quant.location_id.id,
    "lot_id": quant.lot_id.id,
    "package_id": quant.package_id.id,
    "owner_id": quant.owner_id.id,
    "selected": False,
}
```
Why not you assign `line` like this?
IMHO, I think it better.

Reply:
thanks @ps-tubtim, good catch! the code is updated as suggested.